servant-server/src/Servant/Utils/StaticFiles.hs

```diff
@@ -35,10 +36,14 @@ import Filesystem.Path.CurrentOS (decodeString)
 -- behind a /\/static\// prefix. In that case, remember to put the 'serveDirectory'
 -- handler in the last position, because /servant/ will try to match the handlers
 -- in order.
-serveDirectory :: FilePath -> Server Raw
-serveDirectory =
+serveDirectoryWith :: StaticSettings -> ServerT (Raw Application m) n
+serveDirectoryWith settings = Raw (staticApp settings)
+
+serveDirectory :: FilePath -> ServerT (Raw Application m) n
+serveDirectory = serveDirectoryWith . defaultFileServerSettings .
 #if MIN_VERSION_wai_app_static(3,1,0)
-    staticApp . defaultFileServerSettings . addTrailingPathSeparator
+  id .
```

Comment:
Not that is matters much, but couldn't you use `#if !MIN... decodeString .` and not have an `else` clause (and no `id .`)?